snapshot/snapshots/fake_checkouter.go

```diff
@@ -72,6 +73,7 @@ type Initer interface {
 	Init() error
 }
 
+// Creates a checkouter that waits for an Initer to be done Initing before checking out.
 func MakeInitingCheckouter(path string, initer Initer) snapshot.Checkouter {
```

Comment:
We can probably get rid of initingCheckouter... I think it's dead code? (Feel free to TODO)